### PR TITLE
fix: about panel crash

### PR DESCRIPTION
**Pull request by clavin**
#### Description of Change

This code causes a crash bc the `about_panel_options_` gets initialized as a `base::Value`, thus not a `Dict`, and the calls to `GetDict()` fail. This just changes that type to a `base::Value::Dict` and everything 🪄 magically ✨ works. Added a sanity test as a cherry on top. 

**EDIT**: ~~This PR now also unifies every platform by making `showAboutPanel` run asynchronously, where before it varied platform to platform. I think this would be the sensible default for the about panel (fire and forget), but if "run code after closing" is desired, either recommend users to pursue a different solution (e.g. use a BrowserWindow) or we can choose to implement the callback proper later.~~ This will be broken into a separate PR.

#### Release Notes

Notes: Setting the about panel's options no longer crashes.

### CODE REVIEW

**Review by nornagon — APPROVED** [review]

**Review by VerteDinde — APPROVED** [review]

File: shell/browser/browser_linux.cc [hunk 1]
```diff
@@ -162,31 +162,25 @@ bool Browser::IsEmojiPanelSupported() {
 void Browser::ShowAboutPanel() {
   const auto& opts = about_panel_options_;
 
-  if (!opts.is_dict()) {
```
Comment (deepak1556) [inline shell/browser/browser_linux.cc:165]: Current overall change looks good.

To clarify, was the crash only happening on macOS and windows were we don't expect a call to `SetAboutPanelOptions` that actually initializes the value container to hold `Type::Dict` before calling `ShowAboutPanel` ? This check in linux should have avoided the crash as well.

**Comment by MarshallOfSound** [comment]
@deepak1556 It actually crashed inside `SetAboutPanelOptions` --> https://github.com/electron/electron/pull/37373/files#diff-43f417c3f78558ea55277f32e1a078b6bb6c19be7699a2cf61c7020d78f1110dL540 because `GetDict()` was invalid on an uninitialized base value

**Review by deepak1556 — APPROVED** [review]

File: shell/browser/browser_win.cc [hunk 2]
```diff
@@ -743,9 +743,8 @@ void Browser::ShowAboutPanel() {
   dict.SetStringKey("applicationName", exe_info->file_description());
   dict.SetStringKey("applicationVersion", exe_info->product_version());
 
-  if (about_panel_options_.is_dict()) {
-    dict.MergeDictionary(&about_panel_options_);
-  }
+  // Merge user-provided options, overwriting any of the above
+  dict.MergeDictionary(&about_panel_options_);
```
Comment (deepak1556) [inline shell/browser/browser_win.cc:747]:
```suggestion
  dict.GetDict().Merge(about_panel_options_.Clone());
```

**Comment by codebytere** [comment]
@clavin this is stalling/failing on Windows

**Review by codebytere — COMMENTED** [review]
@clavin this is stalling/failing on Windows

**Comment by clavin** [comment]
Yeah I'm trying to get a build going on my Windows machine to figure out why the app is just hanging when it gets to the new tests

**Comment by clavin** [comment]
Apparently `app.showAboutPanel()` is a blocking call on Windows and Linux, but not on macOS  😶

**Comment by codebytere** [comment]
@clavin imo just gate it to linux for now

**Comment by clavin** [comment]
I fixed the issue by making this API non-blocking on all platforms, instead of just on one. That might mean this should have a different semver? To be clear, the docs didn't state the behavior one way or the other, but this still could theoretically cause code relying on the blocking (if there is any???) to execute earlier than intended.

**Comment by clavin** [comment]
I'm going to mark this as needing API review, just in case.

**Comment by deepak1556** [comment]
It would be good to split the async behavior change to a different PR for api-wg review, the dictionary crash fix is good to merge irrespective of it.

**Review by erickzhao — APPROVED** [review]
API LGTM

**Comment by clavin** [comment]
I did both at the same time because I wanted to add a test covering the crash, which is impossible when the method is synchronous :/

But I'm happy to split the test & async change both into a separate PR if it makes things easier :)

**Comment by deepak1556** [comment]
I think the test for `setAboutPanelOptions` is good enough for this PR to cover https://github.com/electron/electron/pull/37373#issuecomment-1439878994, adding test for `showAboutPanel` can be in the async refactor PR.

**Comment by clavin** [comment]
Done 🫡 This PR now only has the crash fix and the API change will be split into a separate PR

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**